src/tokio.rs

```diff
@@ -155,14 +170,24 @@ pub fn init(builder: Builder) {
     *TOKIO_BUILDER.lock().unwrap() = builder
 }
 
+/// Initialize the Tokio runtime with a custom Tokio runtime
+///
+/// Returns Ok(()) if success and Err(()) if it had been inited.
+pub fn init_with_runtime(runtime: &'static TokioRT) -> Result<(), ()> {
+    TOKIO_RUNTIME
+        .set(Runtime::Borrowed(runtime))
+        .map_err(|_| ())
+}
+
 /// Get a reference to the current tokio runtime
-pub fn get_runtime<'a>() -> &'a Runtime {
+pub fn get_runtime<'a>() -> &'a TokioRT {
```

Comment:
I think if we want to add this as a patch release, we'll have to add this as a separate function. Changing the return type is technically a breaking API change even if it will still work in most cases.

We could add a temporary function or just wait until pyo3 v0.18